src/v/cluster/partition_manager.cc

```diff
@@ -61,13 +70,32 @@ ss::future<consensus_ptr> partition_manager::manage(
   raft::group_id group,
   std::vector<model::broker> initial_nodes) {
     gate_guard guard(_gate);
-    bool logs_recovered = co_await maybe_download_log(ntp_cfg);
+    auto dl_result = co_await maybe_download_log(ntp_cfg);
+    auto [logs_recovered, min_kafka_offset, max_kafka_offset, manifest]
+      = dl_result;
     if (logs_recovered) {
         vlog(
           clusterlog.info,
-          "Log download complete, ntp: {}, rev: {}",
+          "Log download complete, ntp: {}, rev: {}, "
+          "min_kafka_offset: {}, max_kafka_offset: {}",
           ntp_cfg.ntp(),
-          ntp_cfg.get_revision());
+          ntp_cfg.get_revision(),
+          min_kafka_offset,
+          max_kafka_offset);
+
+        // Manifest is not empty since we were able to recovery
+        // some data.
+        auto last_segm_it = manifest.rbegin();
```

Comment:
yes, it can't be empty at this point

Reply:
I think it could benefit from an assertion here that it isn't empty. I had to back track through several layers and I think I stopped where target was being populated in download_log from get_offset_map. That seems like something that is too far down the call stack be able to easily maintain (ie know that other users expect that contract). The downside is that this ends up being undefined behavior if there is a mistake on that interface guarantees.

Reply:
I added the assertion